packages/gatsby-theme-newrelic/src/components/GlobalHeader.js

```diff
@@ -52,6 +68,10 @@ const GlobalHeader = ({ editUrl, className, search }) => {
     siteMetadata: { repository },
   } = site;
 
+  window.history.pushState = track(window.history.pushState, () => {
+    setIsQueryChanged(!isQueryChanged);
+  });
```

Comment:
I notice you're not passing any dependencies to `useEffect`. That's fine, but be aware that the callback function to it will be called with every render. Is that intentional?

Reply:
Am I not passing dependencies to `useEffect`? I thought that I was when I pass the array to useEffect which has the dependencies

Reply:
You're right! I think my brain merged this with the `useEffect` call above so I saw it as no dependencies. I see them now! Sorry about that!